exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebUtils.java

```diff
@@ -48,4 +68,87 @@ public static String generateCsrfToken() {
     new SecureRandom().nextBytes(buffer);
     return Base64.getUrlEncoder().withoutPadding().encodeToString(buffer);
   }
+
+  /**
+   * Build an URL of a remote Drillbit endpoint.
+   *
+   * @param work {@link WorkManager} instance needed to retrieve the Drillbit address.
+   * @param request {@link HttpServletRequest} instance needed to to set the URL schema.
+   * @param hostname hostname of the Drillbit.
+   * @param path relative path to the endpoint.
+   * @return remote Drillbit endpoint URL.
+   * @throws RuntimeException if there is no Drillbit at the given hostname.
+   */
+  static URL getDrillbitURL(WorkManager work, HttpServletRequest request, String hostname, String path) {
+    DrillbitEndpoint drillbit = work.getContext().getAvailableBits().stream()
+        .filter(db -> db.getAddress().equals(hostname))
+        .findAny()
+        .orElse(null);
+    if (drillbit == null) {
```

Comment:
This is a programming invariant check. Maybe use wither `assert` or `Preconditions`.

Reply:
Changed to orElseThrow.